src/ActionList/List.tsx

```diff
@@ -124,19 +125,27 @@ export function List(props: ListProps): JSX.Element {
    */
   const renderGroup = (
     groupProps: GroupProps | (Partial<GroupProps> & {renderItem?: typeof Item; renderGroup?: typeof Group})
-  ) => ((('renderGroup' in groupProps && groupProps.renderGroup) ?? props.renderGroup) || Group).call(null, groupProps)
+  ) => {
+    const GroupComponent = (('renderGroup' in groupProps && groupProps.renderGroup) ?? props.renderGroup) || Group
+    return <GroupComponent {...groupProps} key={groupProps.groupId} />
+  }
 
   /**
    * Render an `Item` using the first of the following renderers that is defined:
    * An `Item`-level, `Group`-level, or `List`-level custom `Item` renderer,
    * or the default `Item` renderer.
    */
-  const renderItem = (itemProps: ItemInput, item: ItemInput) =>
-    (('renderItem' in itemProps && itemProps.renderItem) || props.renderItem || Item).call(null, {
-      ...itemProps,
-      sx: {...itemStyle, ...itemProps.sx},
-      item
-    })
+  const renderItem = (itemProps: ItemInput, item: ItemInput) => {
+    const ItemComponent = ('renderItem' in itemProps && itemProps.renderItem) || props.renderItem || Item
+    return (
+      <ItemComponent
+        {...itemProps}
+        key={itemProps.key || uniqueId()}
```

Comment:
I think the simplest solution for now is to allow users to provide a `key` in the item objects that will get passed through, but generate a `uniqueId` if not present. This doesn't add any extra weight to the API but will allow users to take advantage of performance improvements by providing a simple extra bit.

Reply:
We might consider this same approach for dealing with dropdown - we have a default usage that works by accepting memoized objects and comparing object identity, but we can also work with an optional key that users provide to give a bit nicer interface.

Reply:
I think this makes sense as a solution.  I don't know that performance is really that bad without providing a key in this case, so probably won't be something the user even needs to provide.

Regarding DropdownMenu using a similar approach, I'm not sure that would work since they need to be able to pass in the initially selected object using `selectedItem` and they ouldn't have the `uniqueId` at that point.

Reply:
I think this approach is reasonable.

Reply:
`key` feels very React-implementation-detail-specific to me (and, consequently, I’d like for it to not leak into our public API such that consumers must provide it) but the solution here doesn’t require it (i.e. falls back to `uniqueId()`), so I’m on board with this too!